packages/cf.js/API_REFERENCE.md

```diff
@@ -0,0 +1,54 @@
+# API V0.0.2
+## `cf.js`
+
+- `Client`
+    - Properties
+        - `nodeProvider`
+    - Instance methods
+        - `getAppInfos(): AppInfo[]`
+        - `createAppFactory(appDefinition: AppDefinition): AppFactory`
+    - Client lifecycle
+        - `on(eventType, callback)`
+            - EventType: proposeInstall, install, rejectInstall
+- `AppFactory`
+    - Properties
+        - `appDefinition: AppDefinition`
+    - Instance methods
+        - `async proposeInstall(peerAddress, terms): Promise<AppID>`
+        - `async install(appId: AppID): Promise<AppInstance>`
+- `AppInstance`
+    - Properties
+        - `id: AppID` — Identifier for this specific app instance
+        - `definition: AppDefinition`
+    - Instance methods
+        - `async applyAction(action): AppState`
+            - `action`: JSON-encoded object congruent with AppAction struct
+            - Returns ABI decoded representation of the latest signed state of the app.
+        - `async proposeState(state)`: Propose a state to countersign
+        - `async uninstall()`
+            - Uninstall the app
+        - `async getTerms(): [assetType, limit, tokenAddress]`
+    - App lifecycle
+        - `on(eventType, callback)`
+            - EventType: stateUpdate, uninstall
+- `types`
+    - `NodeProvider` (external)
+        - (Injected into webpage)
+        - Instance methods
+            - `postMessage(message)`
+            - `onMessage(callback)`
```

Comment:
Should `onMessage` expect to register a callback for a specific message?

Reply:
I don't know. Doesn't matter from CF.js' perspective, we just consume whatever the node injects. Who owns the node provider and what do they think?

Reply:
Right. Seems like it should, but cc'ing @joelalejandro

Reply:
@cf19drofxots just to clarify, are you imagining something like `onMessage(type, cb)`, so that we'd `onMessage('install', () => {...})` and `onMessage('stateUpdate', () => {...})`?

Reply:
@patience-tema-baron yeah

Reply:
Ping @joelalejandro